tuf/api/metadata.py

```diff
@@ -484,35 +529,61 @@ def __init__(
         version: int,
         spec_version: str,
         expires: datetime,
-        meta: Mapping[str, Any],
+        meta: Mapping[str, MetadataInfo],
     ) -> None:
         super().__init__(_type, version, spec_version, expires)
-        # TODO: Add class for meta
         self.meta = meta
 
+    @classmethod
+    def _get_snapshot_filename(cls, meta_dict: Mapping[str, Any]):
+        """Get the snapshot file name from the metadata dict.
+        The snapshot file could be have a custom extension rather than JSON.
+        See the tuf/api/serialization folder."""
+        meta_files = list(meta_dict.keys())
+        if len(meta_files) != 1:
+            raise ValueError(
+                f"the meta_dict should contain information about the"
+                f"snapshot file only, but got {meta_dict}"
+            )
+        return meta_files[0]
+
     @classmethod
     def from_dict(cls, timestamp_dict: Mapping[str, Any]) -> "Timestamp":
         """Creates Timestamp object from its dict representation. """
         common_args = cls._common_fields_from_dict(timestamp_dict)
-        meta = timestamp_dict.pop("meta")
+        snapshot_filename = cls._get_snapshot_filename(timestamp_dict["meta"])
+        meta = {
+            snapshot_filename: MetadataInfo(
+                **timestamp_dict["meta"][snapshot_filename]
+            )
+        }
         return cls(*common_args, meta)
 
     def to_dict(self) -> Dict[str, Any]:
         """Returns the dict representation of self. """
         timestamp_dict = self._common_fields_to_dict()
-        timestamp_dict.update({"meta": self.meta})
+        snapshot_filename = self._get_snapshot_filename(self.meta)
+        timestamp_dict.update(
+            {
+                "meta": {
+                    snapshot_filename: self.meta[snapshot_filename].to_dict()
+                }
+            }
+        )
```

Comment:
This works but is there a reason to use update over the simpler index access when you only have a single item?
```
 timestamp_dict["meta"] = {
     ...
}
```